src/pages/[gatewayUID]/sensor/[sensorUID]/details.tsx

```diff
@@ -1,330 +1,84 @@
-                        schema={VoltageSensorSchema}
-                      />
-                    ) : (
-                      HISTORICAL_SENSOR_DATA_MESSAGE.NO_VOLTAGE_HISTORY
-                    )}
-                  </Card>
-                </Col>
-                <Col xs={24} sm={24} lg={12}>
-                  <Card className={detailsStyles.sensorChart}>
-                    <h3>Pressure</h3>
-                    <p
-                      data-testid="last-seen-pressure"
-                      className={detailsStyles.sensorChartTimestamp}
-                    >
-                      Last updated {viewModel.sensor.lastActivity}
-                    </p>
-                    {viewModel.readings?.pressure.length ? (
-                      <SensorDetailsChart
-                        label="Pressure"
-                        data={viewModel.readings.pressure}
-                        chartColor="#ffd54f"
-                        schema={PressureSensorSchema}
-                      />
-                    ) : (
-                      HISTORICAL_SENSOR_DATA_MESSAGE.NO_PRESSURE_HISTORY
-                    )}
-                  </Card>
-                </Col>
-              </Row>
-            </TabPane>
-            <TabPane tab="Device Details" key="2">
-              <Form
-                formItems={formItems}
-                onFinish={formOnFinish}
-                onFinishFailed={formOnFinishFailed}
-              />
-            </TabPane>
-          </Tabs>
-        </div>
-      )}
-    </>
+    <LoadingSpinner isLoading={isLoading}>
+      <SensorDetails
+        viewModel={viewModel}
+        err={err || ""}
+        onChangeName={changeName}
+      />
+    </LoadingSpinner>
   );
 };
 
-export default SensorDetails;
+export default SensorDetailsPage;
 
-export const getServerSideProps: GetServerSideProps<SensorDetailsData> =
+export const getServerSideProps: GetServerSideProps<SensorDetailsPageProps> =
   async ({ query }) => {
     const { gatewayUID, sensorUID } = query as SparrowQueryInterface;
     const appService = services().getAppService();
     let viewModel: SensorDetailViewModel = {};
+    let err = "";
 
     try {
       const gateway = await appService.getGateway(gatewayUID);
       const sensor = await appService.getSensor(gatewayUID, sensorUID);
       const readings = await appService.getSensorData(gatewayUID, sensorUID);
       viewModel = getSensorDetailsPresentation(sensor, gateway, readings);
-
-      return {
-        props: { viewModel },
-      };
-    } catch (err) {
-      if (err instanceof Error) {
-        return {
-          props: {
-            viewModel,
-            err: getErrorMessage(err.message),
-          },
-        };
-      }
-      return {
-        props: {
-          viewModel,
-          err: getErrorMessage(ERROR_CODES.INTERNAL_ERROR),
-        },
-      };
+    } catch (e) {
+      err = getErrorMessage(
+        e instanceof Error ? e.message : ERROR_CODES.INTERNAL_ERROR
+      );
```

Comment:
slick 😎

Reply:
Might be worth taking a second to apply this cleanup to the app’s other two pages.